src/flags/duration.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { Flags } from '@oclif/core';
+import { Definition } from '@oclif/core/lib/interfaces';
+import { Messages } from '@salesforce/core';
+import { Duration } from '@salesforce/kit';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/sf-plugins-core', 'messages');
+
+type DurationUnit = Lowercase<keyof typeof Duration.Unit>;
+
+export interface DurationFlagConfig {
+  unit: Required<DurationUnit>;
+  defaultValue?: number;
+  min?: number;
+  max?: number;
+}
+
+/**
+ * Duration flag with built-in default and min/max validation
+ * You must specify a unit
+ * Defaults to undefined if you don't specify a default
+ *
+ * @example
+ * import { SfCommand, buildDurationFlag } from '@salesforce/sf-plugins-core';
+ * public static flags = {
+ *    'wait': buildDurationFlag({ min: 1, unit: , defaultValue: 33 })({
```

Comment:
is it possible to merge the standard flag inputs with the custom ones so that the definition looks like this:

```
wait: buildDurationFlag({
  min: 1,
  unit: Duration.minutes,
  defaultValue: 33,
  char: 'w',
  description: 'Wait time in minutes'
})
```